apps/app_voicemail.c

```diff
@@ -4053,46 +4270,77 @@ static int retrieve_file(char *dir, int msgnum)
 			ast_log(AST_LOG_WARNING, "SQL Describe Column error!\n[%s]\n\n", sql);
 			goto bail_with_handle;
 		}
+
 		if (!strcasecmp(coltitle, "recording")) {
 			off_t offset;
+			char tmp[1] = "";
+
 			res = SQLGetData(stmt, x + 1, SQL_BINARY, rowdata, 0, &colsize2);
 			fdlen = colsize2;
-			if (fd > -1) {
-				char tmp[1] = "";
-				lseek(fd, fdlen - 1, SEEK_SET);
-				if (write(fd, tmp, 1) != 1) {
-					close(fd);
-					fd = -1;
-					continue;
+			ast_trace(-1, "Audio size: %ld\n", colsize2);
+			if (colsize2 == AUDIO_ON_DISK_MAGIC_LEN) {
+				res = SQLGetData(stmt, x + 1, SQL_BINARY, rowdata, AUDIO_ON_DISK_MAGIC_LEN, NULL);
+				if (memcmp(rowdata, AUDIO_ON_DISK_MAGIC, AUDIO_ON_DISK_MAGIC_LEN) != 0) {
+					ast_log(AST_LOG_WARNING, "Invalid audio magic number for '%s'\n", full_fn);
```

Comment:
Could we also print the invalid magic number here?

Reply:
Are we guaranteed that 8 bytes of non-magic isn't perhaps raw audio? (I find it HIGHLY unlikely to be valid, but can it be guaranteed?)

Reply:
@InterLinked1 Yeah, I can print it.

@jkroonza If it's 8 bytes then even if it's raw audio, it's useless.